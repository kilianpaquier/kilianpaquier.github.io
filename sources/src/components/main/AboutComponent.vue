<template>
  <v-container fluid class="pt-0">
    <!-- Title -->
    <h1
      class="h1 font-weight-bold mb-3"
      v-text="$vuetify.lang.t('$vuetify.title')"
    />

    <!-- About paragraph -->
    <p class="text-justify mb-0" v-text="$vuetify.lang.t('$vuetify.about')" />
  </v-container>
</template>

<script lang="ts">
import Component from "vue-class-component";
import Vue from "vue";

@Component
export default class AboutComponent extends Vue {}
</script>
