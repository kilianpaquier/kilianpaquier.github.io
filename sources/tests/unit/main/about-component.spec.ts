import { createLocalVue, mount, Wrapper } from "@vue/test-utils";
import Vuetify from "vuetify";
import Vue from "vue";
import AboutComponent from "@/components/main/AboutComponent.vue";

Vue.use(Vuetify);
Vue.config.silent = true;
Vuetify.config.silent = true;

describe("AboutComponent.vue", () => {
    const localVue = createLocalVue();

    let vuetify: Vuetify;

    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    let wrapper: Wrapper<any>;

    beforeEach(() => {
        vuetify = new Vuetify();

        wrapper = mount(AboutComponent, {
            localVue,
            vuetify,
        });
    });

    it("should create AboutComponent", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
});
