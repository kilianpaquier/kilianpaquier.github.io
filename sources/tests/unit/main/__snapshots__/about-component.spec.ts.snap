// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutComponent.vue should create AboutComponent 1`] = `
<div class="container pt-0 container--fluid">
  <h1 class="h1 font-weight-bold mb-3">$vuetify.title</h1>
  <p class="text-justify mb-0">$vuetify.about</p>
</div>
`;
